apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -582,19 +589,58 @@ public long countBlobs(String containerName, ListContainerOptions options) {
    // copied from BaseBlobStore
    @Beta
    protected String putMultipartBlob(String container, Blob blob, PutOptions overrides) {
-      List<MultipartPart> parts = Lists.newArrayList();
+      ConcurrentLinkedQueue<MultipartPart> parts = new ConcurrentLinkedQueue<MultipartPart>();
+
       long contentLength = checkNotNull(blob.getMetadata().getContentMetadata().getContentLength(),
             "must provide content-length to use multi-part upload");
       MultipartUploadSlicingAlgorithm algorithm = new MultipartUploadSlicingAlgorithm(
             getMinimumMultipartPartSize(), getMaximumMultipartPartSize(), getMaximumNumberOfParts());
       long partSize = algorithm.calculateChunkSize(contentLength);
       MultipartUpload mpu = initiateMultipartUpload(container, blob.getMetadata(), partSize, overrides);
       int partNumber = 1;
+
+      int numThreads = 16;
+      if (System.getProperty("jclouds.mpu.parallel.degree") != null) {
+         numThreads = Integer.getInteger(System.getProperty("jclouds.mpu.parallel.degree"));
```

Comment:
Also I am not very sure if we should be using this property at all and I am not sure about how it is documented (the website states it has a default value of 4, actually, not 16, but it is also not used elsewhere in our code).